### ISSUE TITLE
CBOW: `sigmoid` instead of `softmax` loss with `sparse_categorical_crossentropy`?

### ISSUE THREAD

**Issue by jchwenger** [fchollet/deep-learning-with-python-notebooks#279]
Hi there,

In your [Text Classification](https://github.com/fchollet/deep-learning-with-python-notebooks/blob/master/chapter14_text-classification.ipynb/) chapter, "Pretraining a word embedding", you use the sigmoid as your last activation before a .. `sparse_categorical_crossentropy`?? Looks like the results are the same if you use softmax (some all-too-nice reduction pipeline behind the scene, I'm assuming), but this looks like a typo, no (since all the other models in the chapter have the sigmoid because of the binary sentiment task)? This part in the book also has no comment about this, and Chapter 6 has a neat reminder about the appropriate losses & loss functions, not ideal from a pedagogical point of view?

I hope I haven't overlooked something major, let me know your thoughts !

**Comment by LarsPLoge** [comment]
I also noticed this while preparing for my exam, if the task is classifying the missing word for me it makes much more sense to use the softmax activation to find the most likely missing word.

**Comment by smaeland** [comment]
Hi,
To be more specific, the code in question is listing 14.29 in the book, "Building a CBOW model":
```python
hidden_dim = 64
inputs = keras.Input(shape=(2 * context_size,))
cbow_embedding = layers.Embedding(
    max_tokens,
    hidden_dim,
)
x = cbow_embedding(inputs)
x = layers.GlobalAveragePooling1D()(x)
outputs = layers.Dense(max_tokens, activation="sigmoid")(x)              # <-- should be softmax? 
cbow_model = keras.Model(inputs, outputs)
cbow_model.compile(
    optimizer="adam",
    loss="sparse_categorical_crossentropy",
    metrics=["sparse_categorical_accuracy"],
)
``` 
The other listings are binary classification models, so there the sigmoid makes sense.

**Comment by huanglilong** [comment]
i also think should be softmax  https://github.com/fchollet/deep-learning-with-python-notebooks/issues/292
